_source/_posts/2023-01-23-keycloak-kubernetes-prod.md

```diff
@@ -0,0 +1,637 @@
+
+- `kubernetes.io/ingress.allow-http: "true"`: Required to allow HTTP connections from challenge requests
+- `cert-manager.io/issuer: letsencrypt-staging`: The name of the issuer to acquire the certificate, which must be in the same namespace
+- `acme.cert-manager.io/http01-edit-in-place: "true"`: The ingress is modified in place, instead of creating a new ingress resource for the HTTP01 challenge
+- `cert-manager.io/issue-temporary-certificate: "true"`: A temporary certificate will be set on the secret until the final certificate has been returned. used for keeping compatibility with the `ingress-gce` component.
+
+## Delegate authentication to Auth0
+
+Keycloak supports identity provider federation, meaning it can be configured to delegate authentication to one or more Identity Providers. An example of IDP federation is social login via Facebook or Google. Authentication can be delegated to any IDP supporting OpenID Connect or SAML 2.0. Auth0 uses the OpenID Connect protocol to authenticate users and allows adding custom logic to the login and identity flows via Auth0 Actions. JHipster applications require custom token claims for authorization,  and these can be configured using Auth0 Actions. The sections below describe the step-by-step process to add Auth0 authentication to the gateway using Keycloak, without making any application changes. You can find a detailed description of the [brokering flow](https://www.keycloak.org/docs/latest/server_admin/#_identity_broker_overview) in the Keycloak server administration guide.
+
+### Create an Auth0 account
+
+Sign up at [Auth0](https://auth0.com/signup) and install the [Auth0 CLI](https://github.com/auth0/auth0-cli). Then run:
+
+```shell
+auth0 login
+```
+
+The terminal will display a device confirmation code and open a browser session to activate the device.
+
+**NOTE**: My browser was not displaying anything, so I had to manually activate the device by opening the URL `https://auth0.auth0.com/activate?user_code={deviceCode}`.
+
+On successful login, you will see the tenant, which you will use as issuer later:
+
+```
+✪ Welcome to the Auth0 CLI 🎊
+
+If you don't have an account, please create one here: https://auth0.com/signup.
+
+Your device confirmation code is: KGFL-LNVB
+
+ ▸    Press Enter to open the browser to log in or ^C to quit...
+
+Waiting for the login to complete in the browser... ⣻Opening in existing browser session.
+Waiting for the login to complete in the browser... done
+
+ ▸    Successfully logged in.
+ ▸    Tenant: dev-avup2laz.us.auth0.com
+```
+
+The next step is to create a client app:
+
+```shell
+auth0 apps create
+```
+
+When prompted, choose the following options:
+
+- Name: **jhipster**
+- Description: **<optional description>**
+- Type: **Regular Web Application**
+- Callback URLs: **https://keycloak.\<namespace\>.\<public-ip\>.nip.io/login/oauth2/code/auth0**
```

Comment:
```suggestion
- Callback URLs: **https://keycloak.\<namespace\>.\<public-ip\>.nip.io/login/oauth2/code/auth0**
```

I believe this URL should end in `oidc`, not `auth0`.

Reply:
I believe the suffix must be the alias given to the provider in Keycloak. Will add a note anc double check.

Reply:
The url is updated to match the alias after saving the oidc provider